grype/db/curator.go

```diff
@@ -267,6 +278,24 @@ func (c *Curator) download(listing *ListingEntry, downloadProgress *progress.Man
 	return tempDir, nil
 }
 
+func (c *Curator) validateStaleness(m *Metadata) error {
+	if m == nil {
+		return nil
+	}
+
+	age := time.Since(m.Built).Round(time.Hour)
+	// dataStalenessLimit == 0 means to not error on staleness
+	if c.dataStalenessLimit > 0 && age > c.dataStalenessLimit {
+		return fmt.Errorf("%w: last updated %s ago (> threshold %s)", ErrDataIsStale, age, c.dataStalenessLimit)
+	}
+
+	if age > DefaultStalenessThreshold {
+		log.Warnf("your vuln db was updated %s ago and it might generate wrong results. To update run: grype db update", age)
```

Comment:
This message could be improved to be more clear and idiomatic to our log messages:

1. The problem isn't when the database was updated, it's when it was built.
2. I'd say "inaccurate" instead of something as simplified as "wrong".
3. We typically don't say "your vuln db". Instead we tend to say "the vulnerability database".

Reply:
Thank you for the language clarifications, I applied the suggestions.